pyanaconda/ui/tui/spokes/software_selection.py

```diff
@@ -93,6 +100,11 @@ def _initialize(self):
         """Initialize the spoke in a separate thread."""
         threadMgr.wait(THREAD_PAYLOAD)
 
+        self.available_kernels = get_display_kernel_features(self.payload)
+        self.kernel_selection = dict.fromkeys(self.available_kernels, False)
+        # kernel-redhat should be the default when available
+        self.kernel_selection["upstream"] = True
```

Comment:
Oh, that was really bad, I am sorry this got through, thanks for catching this.

Reply:
This seems to be still true. In this commit nor the previous ones I don't see any "upstream" kernel implementation. Also it should be renamed because it's not called upstream anywhere else.